File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[]res = new int[n];
        for(int i=0;i<n;i++){
             int diff=0;
            for(int j=i+1;j<n;j++){
               
                if(temperatures[j]>temperatures[i]){
                   diff = j-i;
                   break;
                }

            }
            res[i] = diff;
        }

        return res;
    }
}

File: Data Structures & Algorithms/decode-string/submission-0.swift
class Solution {
    func decodeString(_ s: String) -> String {

        let n = s.count

        var st: [Character] = []

        if n == 0 {
            return " "
        }

        let chars = Array(s)

        for ch in chars{

            if ch != "]" {
                st.append(ch)
            } else{

            var str = ""

           while let last = st.last, last != "[" {
                str = String(st.removeLast()) + str

            }

            st.removeLast() //remove the [

            //now getting len

            var numStr = ""

            while let last = st.last, last.isNumber {
                numStr = String(st.removeLast()) + numStr

            }

            let res = String(repeating:str, count: Int(numStr) ?? 0)

           

            for letter in res {
                st.append(letter)
            }

            }

        }


        return String(st)

        



    }
}
